crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -95,28 +99,57 @@ async fn app_validation_workflow_inner(
 ) -> WorkflowResult<WorkComplete> {
     let env = workspace.validation_limbo.env().clone();
 
-    let ops: Vec<ValidationLimboValue> = fresh_reader!(env, |r| workspace
-        .validation_limbo
-        .drain_iter_filter(&r, |(_, vlv)| {
-            match vlv.status {
-                // We only want sys validated or awaiting app dependency ops
-                ValidationLimboStatus::SysValidated | ValidationLimboStatus::AwaitingAppDeps(_) => {
-                    Ok(true)
-                }
-                ValidationLimboStatus::Pending | ValidationLimboStatus::AwaitingSysDeps(_) => {
-                    Ok(false)
-                }
-            }
-        })?
-        // TODO: Sort into min heap like sys val
-        .collect())?;
+    // Drain the ops into a sorted binary heap
+    let sorted_ops: BinaryHeap<std::cmp::Reverse<OrderedOp<ValidationLimboValue>>> =
```

Comment:
Interesting to see the use of BinaryHeaps showing up. Seems like this is the preferred technique for building a sorted list and iterating over it? Is it because they're quicker to construct vs BTreeSet? And then it's just as fast to iterate over?

Reply:
Yes my understanding is binary heaps give you a quicker sort on insert and then iteration when you don't need to look up via hash. They are just priority queues which seems to suit this purpose pretty exactly